mkfat32.zig

```diff
@@ -0,0 +1,361 @@
+const std = @import("std");
+const File = std.fs.File;
+
+/// A FAT32 static structure for creating a empty FAT32 image. This contains helper functions
+/// for creating a empty FAT32 image.
+/// For more information, see: https://en.wikipedia.org/wiki/Design_of_the_FAT_file_system
+pub const Fat32 = struct {
+    /// The struct for creating the FAT32 serial number. It is based on the current time.
+    const SerialTime = struct {
+        year: u16,
+        month: u8,
+        day: u8,
+        hour: u8,
+        minute: u8,
+        second: u8,
+        millisecond_10: u8,
+    };
+
+    /// The FAT32 boot sector header (without the jmp and boot code) including the extended
+    /// parameter block for FAT32 with the extended signature 0x29.
+    pub const Header = struct {
+        /// The OEM.
+        oem: [8]u8 = "ZystemOS".*,
```

Comment:
It's almost as if we planned the name to be the exact length needed :D

Reply:
I know right :P